src/main/java/com/dataguise/saas/service/SchedulerService.java

```diff
@@ -0,0 +1,434 @@
+            logger.error("Schedule name is missing.");
+            throw new BadRequestException(ErrorConstants.ERROR_VALIDATING_SCHEDULER_NAME.getErrorMessage(), HttpStatus.BAD_REQUEST);
+        }
+
+        if (schedulerDto.getScanType() == null) {
+            logger.error("Scan Type is a required parameter and possible values are Weekly/Daily");
+            throw new BadRequestException(ErrorConstants.ERROR_VALIDATING_SCAN_TYPE.getErrorMessage(), HttpStatus.BAD_REQUEST);
+        }
+        if (schedulerDto.getStartDateTime() == null) {
+            logger.error("Start date and time cannot be empty.");
+            throw new BadRequestException(ErrorConstants.ERROR_VALIDATING_START_DATE.getErrorMessage(), HttpStatus.BAD_REQUEST);
+        }
+
+        if (scheduleId != null) {
+            List<SchedulerDTO> existingScheduleList = getSchedule(Optional.of(scheduleId));
+            DgTaskScheduleInfoBean existingSchedule;
+            if (!existingScheduleList.isEmpty()) {
+                existingSchedule = existingScheduleList.get(0).getSchedule();
+            } else {
+                throw new BadGatewayException(ErrorConstants.ERROR_GETTING_SCHEDULER_BY_ID.getErrorMessage(), HttpStatus.BAD_GATEWAY);
+            }
+            logger.info("Existing task info bean for schedule:" + existingSchedule);
+            formatScheduleDTO(schedulerDto);
+            logger.info("Formatted schedule DTO from UI: " + schedulerDto);
+
+            List<Integer> taskIdList = new ArrayList<>();
+            taskIdList.add(existingSchedule.getTaskId());
+
+            //existingSchedule.setId(0);
+            existingSchedule.setScheduleName(schedulerDto.getScheduleName());
+            existingSchedule.setScheduleInterval(schedulerDto.getScheduleInterval());
+            if (schedulerDto.getSourceModule() != null &&
+                    schedulerDto.getSourceModule().equalsIgnoreCase(Modules.S3LFA.getModuleType()))
+                existingSchedule.setTaskType(taskBuilderService.DETECTION_TYPE);  // In getScheduler call taskType comes as 'file' from controller backend.
+
+            //existingSchedule.setTaskType(schedulerDto.getTaskType());
+            existingSchedule.scheduleScanStartDate(schedulerDto.getStartDateTime());
+            existingSchedule.setScheduleScanEndDate(SCHEDULE_END_DATE);
+            existingSchedule.setScheduleType(schedulerDto.getScanType());
+            existingSchedule.setTaskId(null);
+            existingSchedule.setExecutionType(null);
+            existingSchedule.setTaskIds(taskIdList);
+            existingSchedule.setScheduleDay(schedulerDto.getScheduleDay());
+            existingSchedule.setTimeZone("UTC");
+            //By default it is set to true no matter what we pass.
+            //existingSchedule.setEnabled(true);
+
+            logger.info("Updated existing task info bean to controller: " + existingSchedule);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/SchedulerService.java L307
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1355947404>*